torch/optim/optimizer.py

```diff
@@ -233,17 +233,73 @@ class Optimizer:
     _optimizer_step_post_hooks: Dict[int, OptimizerPostHook]
     _optimizer_state_dict_pre_hooks: 'OrderedDict[int, Callable[["Optimizer"], None]]'
     _optimizer_state_dict_post_hooks: 'OrderedDict[int, Callable[["Optimizer", StateDict], Optional[StateDict]]]'
-    _optimizer_load_state_dict_pre_hooks: 'OrderedDict[int, Callable[["Optimizer", StateDict], Optional[StateDict]]]'
+    # _optimizer_load_state_dict_pre_hooks handled by @property
     _optimizer_load_state_dict_post_hooks: 'OrderedDict[int, Callable[["Optimizer"], None]]'
 
+
+    @staticmethod
+    def _verify_param_groups_align(groups, saved_groups) -> Dict[int, torch.Tensor]:
+        """_verify_param_groups_align is a helper method that verifies the param groups of
+        a saved checkpoint and the current optimizer are aligned and then returns an
+        id map mapping param ID (integer) to a parameter.
+        """
+
+        if len(groups) != len(saved_groups):
+            raise ValueError("loaded state dict has a different number of "
+                             "parameter groups")
```

Comment:
Could you add the relevant length to the error message to prevent someone going crazy later :P